### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Ders1k**

### CODE REVIEW

File: src/auth/auth.controller.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    return this.authService.signIn(payload);
+  }
+
+  @Post('/signup')
+  async signUp(@Body() payload: signUpDto) {
+    const candidate = this.userService.getUserByEmail(payload.email);
+    if (candidate) {
+      throw new ConflictException('User already exist');
+    }
+    this.userService.createUser(payload);
```
Comment (nottoday2471) [inline src/auth/auth.controller.ts:43]: should be better to add some errors catch

File: src/auth/dto/index.ts [hunk 2]
```diff
@@ -0,0 +1,2 @@
+export * from './signin.dto';
+export * from './signup.dto';
```
Comment (nottoday2471) [inline src/auth/dto/index.ts:2]: 👀

File: src/auth/dto/signin.dto.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { IsEmail, IsNotEmpty } from 'class-validator';
+
+export class signInDto {
+  @IsEmail()
+  @IsNotEmpty()
+  email: string;
+
+  @IsNotEmpty()
+  password: string;
```
Comment (nottoday2471) [inline src/auth/dto/signin.dto.ts:9]: maybe add here string validation

File: src/auth/dto/signup.dto.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { IsEmail, IsNotEmpty } from 'class-validator';
+
+export class signUpDto {
+  @IsNotEmpty()
+  name: string;
```
Comment (nottoday2471) [inline src/auth/dto/signup.dto.ts:5]: same

File: src/auth/dto/signup.dto.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+  @IsNotEmpty()
+  surname: string;
+
+  @IsEmail()
+  @IsNotEmpty()
+  email: string;
+
+  @IsNotEmpty()
+  password: string;
```
Comment (nottoday2471) [inline src/auth/dto/signup.dto.ts:15]: same

**Comment by nottoday2471** [comment]
LGTM

File: src/auth/dto/signup.dto.ts (hunk 4 above)
Comment (beastmode2306) [reply src/auth/dto/signup.dto.ts:6]: follow the same naming style over all dto classes

File: src/device/device.controller.ts [hunk 6]
```diff
@@ -35,8 +35,26 @@ export class DeviceController {
     return device;
   }
 
+  @Get('/brand/:id')
+  async getDevicesByBrand(@Param('id', ParseIntPipe) id: number) {
+    const devices = await this.deviceService.getDeviceByBrand(id);
+    if (devices.length < 1) {
```
Comment (nottoday2471) [inline src/device/device.controller.ts]: 👀

**Comment by nottoday2471** [comment]
GJ

**Pull request merged**